apisix/admin/consumers.lua

```diff
@@ -102,6 +103,33 @@ function _M.get(consumer_name)
     end
 
     utils.fix_count(res.body, consumer_name)
+
+    if consumer_name then
+        -- if data is queried for a single consumer, and there is any plugin where the vault config
+        -- is enabled - it fetches vault data and returns combined with etcd response.
```

Comment:
IMHO, it is not a good idea to merge the vault data with etcd response. Now everyone who can access Admin API could access the vault data, which is against the reason to use vault.

Reply:
That's actually true. What do you suggest here, removing the part or supporting the feature in a different way?
One reason to have this feature:
- have a broader look of the data that are being used for the consumer. Instead of looking the responses of etcd and vault separately. But yes your concern is totally apt.

Reply:
I think we can remove it.

BTW, when people store their data in protection storage, easier to access or can be written is no longer a plus for them. Restriction access is more helpful.

Reply:
Ack